metadata-ingestion/src/datahub/ingestion/source/redash.py

```diff
@@ -0,0 +1,466 @@
+        self.client.session.headers.update(
+            {
+                "Content-Type": "application/json",
+                "Accept": "application/json",
+            }
+        )
+
+        self.api_page_limit = self.config.api_page_limit or math.inf
+
+        # Test the connection
+        test_response = self.client._get(f"{self.config.connect_uri}/api")
+        if test_response.status_code == 200:
+            logger.info("Redash API connected succesfully")
+            pass
+
+            # # Only for getting data source types available in Redash
+            # # We use this as source for `REDASH_DATA_SOURCE_TO_DATAHUB_MAP` variable
+            # data_source_types = self.client._get(f"/api/data_sources/types").json()
+            # for dst in data_source_types:
+            #     print(f'\'{dst["type"]}\'', ":", {"platform": None, }, ",")
+
+    @classmethod
+    def create(cls, config_dict: dict, ctx: PipelineContext) -> Source:
+        config = RedashConfig.parse_obj(config_dict)
+        return cls(ctx, config)
+
+    @lru_cache(maxsize=None)
+    def _get_chart_data_source(self, data_source_id):
+        url = f"/api/data_sources/{data_source_id}"
+        resp = self.client._get(url).json()
+        return resp
+
+    @lru_cache(maxsize=None)
+    def _get_datasource_urn_from_data_source_id(self, data_source_id):
+        data_source = self._get_chart_data_source(data_source_id)
+        data_source_type = data_source.get("type")
+        data_source_name = data_source.get("name")
+        data_source_options = data_source.get("options", {})
+
+        if data_source_type:
+            map = REDASH_DATA_SOURCE_TO_DATAHUB_MAP.get(
+                data_source_type, {"platform": DEFAULT_DATA_SOURCE_PLATFORM}
+            )
+            platform = map.get("platform")
+            platform_urn = f"urn:li:dataPlatform:{platform}"
+
+            db_name_key = map.get("db_name_key", "db")
+            db_name = data_source_options.get(db_name_key, DEFAULT_DATA_BASE_NAME)
+
+            # Redash Query Results
+            if data_source_type == "results":
+                dataset_urn = (
+                    f"urn:li:dataset:("
+                    f"{platform_urn},{data_source_name},{self.config.env})"
+                )
+                return data_source_type, dataset_urn
+
+            # Other Redash supported data source as in REDASH_DATA_SOURCE_TO_DATAHUB_MAP
+            if db_name:
+                dataset_urn = (
+                    f"urn:li:dataset:(" f"{platform_urn},{db_name},{self.config.env})"
+                )
+                return data_source_type, dataset_urn
+        return data_source_type, None
+
+    def _get_dashboard_description_from_widgets(
+        self, dashboard_widgets: List[Dict]
+    ) -> str:
+        description = ""
+
+        for widget in dashboard_widgets:
+            visualization = widget.get("visualization")
+            if not visualization:
```

Comment:
If you're checking if `visualization` is `None` here, you may want to use `if visualization is None` instead: https://stackoverflow.com/questions/7816363/if-a-vs-if-a-is-not-none